packages/application-components/src/components/modal-pages/form-modal-page/form-modal-page.js

```diff
@@ -0,0 +1,96 @@
+import { ContentWrapper } from '../internals/modal-page.styles';
+import messages from '../internals/messages';
+
+const getFormattedLabel = (label, intl) =>
+  typeof label === 'string' ? label : intl.formatMessage(label);
+
+const FormModalPage = props => (
+  <ModalPage
+    level={props.level}
+    title={props.title}
+    isOpen={props.isOpen}
+    zIndex={props.zIndex}
+    onClose={props.onClose}
+    baseZIndex={props.baseZIndex}
+    parentSelector={props.parentSelector}
+  >
+    <ModalPageTopBar
+      onClose={props.onClose}
+      currentPathLabel={props.topBarCurrentPathLabel || props.title}
+      previousPathLabel={props.topBarPreviousPathLabel}
+    />
+    <ModalPageHeader
+      title={props.title}
+      subtitle={props.subtitle}
+      customControls={props.customControls}
+      labelPrimaryButton={getFormattedLabel(
+        props.labelPrimaryButton,
+        props.intl
+      )}
+      labelSecondaryButton={getFormattedLabel(
+        props.labelSecondaryButton,
+        props.intl
+      )}
+      onPrimaryButtonClick={props.onPrimaryButtonClick}
+      onSecondaryButtonClick={props.onSecondaryButtonClick}
+      isPrimaryButtonDisabled={props.isPrimaryButtonDisabled}
+      dataAttributesPrimaryButton={props.dataAttributesPrimaryButton}
+      dataAttributesSecondaryButton={props.dataAttributesSecondaryButton}
+    />
+    <ContentWrapper>{props.children}</ContentWrapper>
+  </ModalPage>
+);
+FormModalPage.displayName = 'FormModalPage';
+FormModalPage.propTypes = {
+  level: PropTypes.number,
+  title: PropTypes.string.isRequired,
+  zIndex: PropTypes.number,
+  isOpen: PropTypes.bool.isRequired,
+  onClose: PropTypes.func,
+  children: PropTypes.node.isRequired,
+  baseZIndex: PropTypes.number,
+  parentSelector: PropTypes.string,
+  // TopBar props
+  topBarCurrentPathLabel: PropTypes.string,
+  topBarPreviousPathLabel: PropTypes.string,
+  // Header props
+  subtitle: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  customControls: PropTypes.node,
+  labelSecondaryButton: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      defaultMessage: PropTypes.string.isRequired,
+    }),
+  ]).isRequired,
+  labelPrimaryButton: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      defaultMessage: PropTypes.string.isRequired,
+    }),
+  ]).isRequired,
+  onPrimaryButtonClick: PropTypes.func.isRequired,
+  onSecondaryButtonClick: PropTypes.func,
+  isPrimaryButtonDisabled: PropTypes.bool,
+  dataAttributesPrimaryButton: PropTypes.object,
+  dataAttributesSecondaryButton: PropTypes.object,
+  // injected
+  intl: PropTypes.shape({
+    formatMessage: PropTypes.func.isRequired,
+  }).isRequired,
+};
+FormModalPage.defaultProps = {
+  labelPrimaryButton: messages.confirm,
+  labelSecondaryButton: messages.cancel,
+};
+
+FormModalPage.Intl = { confirm: messages.confirm, cancel: messages.cancel };
```

Comment:
How about we split the messages into 2 files?
- `button-messages.js`
- `messages.js`

Then we can simply `FormModalPage.Intl = buttonMessages;`

Reply:
I considered it, but preferred this solution to prevent having too many files. But, I have no strong opinion one way or another, so if you prefer to split we can do that.

Reply:
I fear that we forget to update the list here if we add more messages.

Also, we should reuse the same messages that we had in dialogs, to avoid translating them again.

Reply:
Here you go: https://github.com/commercetools/merchant-center-application-kit/pull/642/commits/19a1ed07dada5b647f1881c4084647a66975216d

Reply:
Thanks 😄